lib/Genesis/CI/Legacy.pm

```diff
@@ -1119,8 +1121,8 @@ EOF
             CACHE_DIR:            $alias-cache
             GIT_BRANCH:           (( grab pipeline.git.branch ))
             GIT_PRIVATE_KEY:      (( grab pipeline.git.private_key ))
-            VAULT_ROLE_ID:        $pipeline->{pipeline}{vault}{role}
-            VAULT_SECRET_ID:      $pipeline->{pipeline}{vault}{secret}
+            VAULT_ROLE_ID:        (( grab pipeline.vault.role ))
+            VAULT_SECRET_ID:      (( grab pipeline.vault.secret ))
```

Comment:
Since we're no longer using these, we should remove all the other references to handling them (in other parts of Legacy.pm), and warn if they are present.

Reply:
I've kept them here in case an operator wanted to override what we do. i.e. a security team doesn't like our policies or whatnot, and instead they'd prefer to manage their own AppRole/policies. By default, if `pipeline.vault.*` in `ci.yml` takes precedent if they're defined. Otherwise, fallback to our in-Vault, Concourse-kit-generated credentials.

Reply:
love it!